src/react-components/code-sample.tsx

```diff
@@ -33,59 +41,79 @@ export type CodeSamplesViewerProps = {
    * the component. Receives data about the selected code sample. The library
    * comes pre-packaged with some sensible options.
    *
-   * @see CodeSampleMethodTitle
+   * @see CodeSampleTitle
    * @see CodeSampleFilenameTitle
    * @default CodeSampleMethodTitle
    */
-  title?: CodeSampleTitleComponent | React.ReactNode | string;
-  /** The operation to get a code sample for. Can be queried by either
-   * operationId or method+path.
+  title?: CodeSampleTitleComponent | React.ReactNode | string | "none";
+  /** The operations to get code samples for. If only one is provided, no selector will be shown.
+   * Can be queried by either operationId or method+path.
    */
-  operation: MethodPaths | OperationId;
+  operations?: MethodPaths[] | OperationId[];
   /**
    * Optional client. Use this if the component is being used outside of
    * SpeakeasyCodeSamplesContext.
    */
   client?: SpeakeasyCodeSamplesCore;
+  /**
+   * Sets the style of the code window.
+   */
+  codeWindowStyle?: React.CSSProperties;
+
   className?: string | undefined;
   style?: React.CSSProperties;
 };
 
 export function CodeSamplesViewer({
   theme = "system",
-  className,
-  title,
-  operation,
-  style,
+  title = CodeSampleFilenameTitle,
+  defaultLanguage,
+  defaultOperation,
+  operations,
   copyable,
   client: clientProp,
+  style,
+  codeWindowStyle,
+  className,
 }: CodeSamplesViewerProps) {
-  const request: GetCodeSamplesRequest = React.useMemo(() => {
-    if (typeof operation === "string") return { operationIds: [operation] };
-    return { methoPaths: [operation] };
-  }, [operation]);
+  const requestParams: GetCodeSamplesRequest = React.useMemo(() => {
+    if (typeof operations?.[0] === "string")
+      return { operationIds: operations as OperationId[] };
+    else if (operations?.[0]?.method && operations[0].path)
+      return { methodPaths: operations as MethodPaths[] };
+
+    return {};
+  }, [operations]);
 
-  const { state, setSelectedLanguage } = useCodeSampleState({
+  const { state, selectSnippet } = useCodeSampleState({
     client: clientProp,
-    requestParams: request,
+    requestParams,
   });
 
+  // On mount, select the defaults
+  useEffect(() => {
+    if (!state.snippets || state.status !== "success") return;
+    selectSnippet({ language: defaultLanguage, operationId: defaultOperation });
+  }, [state.status]);
+
   const systemColorMode = useSystemColorMode();
   const codeTheme = React.useMemo(() => {
     if (theme === "system") return codehikeTheme[systemColorMode];
     return codehikeTheme[theme];
   }, [theme, systemColorMode]);
 
-  const longestCodeHeight = React.useMemo(() => {
```

Comment:
wanted to comment to verify this is something we want to omit. The moonshine component calculates height to prevent layout shift as samples w/ differing heights are swapped, but also allows the user to provide a max height. That approach ensures the previewer is consistently the same height.

Reply:
Maybe I'll put this behind a prop like `fixedHeight`